lib/parsers/acorn-extensions.js

```diff
@@ -106,23 +106,33 @@ function parseExport(node, exports) {
   return this.finishNode(node, "ExportNamedDeclaration");
 }
 
-function lookAhead(n) {
+// Calls the given callback with the state of the parser temporarily
+// advanced by calling this.nextToken() n times, then rolls the parser
+// back to its original state and returns whatever the callback returned.
+function withLookAhead(n, callback) {
   const old = Object.assign(Object.create(null), this);
   while (n-- > 0) this.nextToken();
-  const copy = Object.assign(Object.create(null), this);
-  Object.assign(this, old);
-  return copy;
+  try {
+    return callback.call(this);
+  } finally {
```

Comment:
If it's just `() => parser.type` you can optimize the callback call without the `this` binding and just do
```
return callback();
```